stagecraft/apps/datasets/views/data_set.py

```diff
@@ -1,44 +1,201 @@
+    generated_id = False
+    schema = {
+        "$schema": "http://json-schema.org/schema#",
+        "type": "object",
+        "properties": {
+            "bearer_token": {
+                "type": "string"
+            },
+            "capped_size": {
+                "type": "string"
+            },
+            "name": {
+                "type": "string"
+            },
+            "data_type": {
+                "type": "string"
+            },
+            "realtime": {
+                "type": "boolean"
+            },
+            "auto_ids": {
+                "type": "string"
+            },
+            "queryable": {
+                "type": "boolean"
+            },
+            "upload_format": {
+                "type": "string"
+            },
+            "published": {
+                "type": "boolean"
+            },
+            "upload_filters": {
+                "type": "string"
+            },
+            "max_age_expected": {
+                "type": "number"
+            },
+            "data_group": {
+                "type": "string"
+            },
+            "raw_queries_allowed": {
+                "type": "boolean"
+            },
+        },
+        "required": ["data_type", "data_group"],
+        "additionalProperties": False,
+    }
 
-    return HttpResponse(json_str, content_type='application/json')
+    @method_decorator(permission_required('signin'))
+    @method_decorator(never_cache)
+    @method_decorator(vary_on_headers('Authorization'))
+    def get(self, user, request, **kwargs):
+        name = kwargs.get(self.id_field, None)
+        if name is not None:
+            try:
+                data_set = DataSet.objects.get(name=name)
+                user_is_not_admin = 'admin' not in user['permissions']
+                user_is_not_assigned = data_set.backdropuser_set.filter(
+                    email=user['email']).count() == 0
+                if user_is_not_admin and user_is_not_assigned:
+                    logger.warn("Unauthorized access to '{}' by '{}'".format(
+                        name, user['email']))
+                    raise DataSet.DoesNotExist()
+            except DataSet.DoesNotExist:
+                error = {'status': 'error',
+                         'message': "No Data Set named '{}' exists".
+                         format(name)}
+                logger.warn(error)
+
+                error["errors"] = [
+                    create_error(request, 404, detail=error['message'])]
+
+                return HttpResponseNotFound(to_json(error))
+        else:
+            return self.list(user, request)
+
+        json_str = to_json(data_set.serialize())
+
+        return HttpResponse(json_str, content_type='application/json')
+
+    # No longer called directly - happens through get which is protected.
+    def list(self, user, request):
+        # 400 if any query string keys were not in allowed set
+        if not set(request.GET).issubset(self.list_filters):
```

Comment:
It might be good to move this into ResourceView as it strikes me as useful for other subclasses